### PR TITLE
Component Replacement 

### PR DESCRIPTION
**Pull request by RayRedGoose**
### This PR contains the next changes:

- Replaced App, DisplayInsights, FinalIdeaField, Header, SecretSauce, Timer, UserProfile components to Containers folder because they have a connection with Redux store.
- Replaced WelcomePage, IdeaEventHorizon, SignUpForm components to Components folder because they are not connected with Redux store.
- Added redirection to Welcome page from other URLs if there is no user logged.
- Removed unused variables.
- Added missed type checks.

### CODE REVIEW

**Comment by JamesRexMiller4** [comment]
Hey I noticed that the insights div I was working on in RoundTwo still has some buggy styles. Particularly if you hold down one letter on the keyboard with no spaces, it will break out of the containing div width. Can you take a look at that when you get a chance?

**Comment by JamesRexMiller4** [comment]
Also what do you think about allowing the typed input to remain as it goes from round 1 to 2, 2 to 3 for RoundTwo. Part of me was thinking of clearing the input automatically, but then another part of me was thinking that if they had typed up a question they really liked and then it was removed that might be worse than them having to delete it themselves

**Comment by RayRedGoose** [comment]
I found some more bugs in our app and fixed them:
- after login is clicked and app fetches user and data, there is a possibility to start signup at the same time. 
- your issue with questions in round 2 also exists in a final round. I fixed it by adding hyphens and `overflow-wrap: brake-word`.
- duplicated components with the same content but different names. I deleted an unused component.

**Comment by RayRedGoose** [comment]
Also, I updated snapshots for failed tests.

**Comment by RayRedGoose** [comment]
> Also what do you think about allowing the typed input to remain as it goes from round 1 to 2, 2 to 3 for RoundTwo. Part of me was thinking of clearing the input automatically, but then another part of me was thinking that if they had typed up a question they really liked and then it was removed that might be worse than them having to delete it themselves

I added clear input functionality to useEffect before it's going to the next sentence. Now, we have empty input for every new sentence.

File: src/Components/SignUpForm/SignUpForm.test.tsx [hunk 1]
```diff
@@ -3,83 +3,62 @@
...
+  //
+  //       wrapper.find('input').first().simulate('change', mockEvent);
+  //       wrapper.find('#last-name').first().simulate('change', mockEvent2);
+  //       wrapper.find('#email').first().simulate('change', mockEvent3);
+  //       wrapper.find('#password').first().simulate('change', mockEvent4);
+  //       wrapper.find('input').last().simulate('change', mockEvent5);
+  //
+  //       expect(wrapper.find('button').getDOMNode().disabled).toEqual(false);
+  //   });
+  // });
```
Comment (JamesRexMiller4) [inline src/Components/SignUpForm/SignUpForm.test.tsx]: I found some commented out code when I redid the ActionsField component on my last PR. I am cool with merging this in, but lets maybe take a day to go through and refactor our tests, and make sure every component has at least one test for it

File: src/Components/BrainstormCard/BrainstormCard.test.tsx [hunk 2]
```diff
@@ -8,10 +8,16 @@ describe('BrainstormCard component', () => {
   const mockProps = {
     id: 1,
```
Comment (JamesRexMiller4) [inline src/Components/BrainstormCard/BrainstormCard.test.tsx]: Is this Prettier making these changes? I don't mind reviewing files, but this seems a little superfluous

File: src/Components/GenerateInsights/GenerateInsights.scss [hunk 3]
```diff
@@ -20,6 +20,12 @@
   h3 {
     font-size: 1rem;
     text-align: left;
+    -webkit-hyphens: auto;
+    -moz-hyphens: auto;
+    -ms-hyphens: auto;
+    hyphens: auto;
+    overflow-wrap: break-word;
+    width: 90%;
```
Comment (JamesRexMiller4) [inline src/Components/GenerateInsights/GenerateInsights.scss]: Amazo! 🚀

File: src/Components/LoginForm/LoginForm.test.tsx [hunk 4]
```diff
@@ -4,88 +4,62 @@
...
+    setIsLoading: mockSetIsLoading,
+    setCookie: mockSetCookie
+  };
 
   afterEach(() => {
     jest.clearAllMocks();
   });
 
-
   it('should match the snapshot', () => {
```
Comment (JamesRexMiller4) [inline src/Components/LoginForm/LoginForm.test.tsx]: Nice refactor, I have been thinking afterEach is more important than beforeEach

**Comment by JamesRexMiller4** [comment]
Nice job reorganizing the files and refactoring here and there!

**Pull request merged**